### PR TITLE
Deletion - fix ordering bug

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
In some complex deletion cases, we must delete attribute ownerships and role playing first, followed by relations instances and other instances. We applied an incorrect tie breaker, resulting in deleting relations before attribute ownerships, resulting in `vertex deleted` exceptions from Janus. We also enforce deleting attribute ownerships before role players, as role player deletion could lead to cascading deletions.

## What are the changes implemented in this PR?
* Add explicit ordering of deletes: HAS, ROLE_PLAYER, RELATION, NON_RELATION

### CODE REVIEW

**Comment by flyingsilverfin** [comment]
@jmsfltchr this is failing because of explanation tests! Are you going to be able to submit your PR with fixes for these to master anytime soon?

**Comment by jmsfltchr** [comment]
Aha yes @flyingsilverfin we can put that in soon

File: graql/executor/property/HasAttributeExecutor.java [hunk 1]
```diff
@@ -135,20 +134,24 @@ public VarProperty property() {
...
+            vars.add(var);
+            vars.add(property.attribute().var());
+            return vars;
         }
 
         @Override
         public TiebreakDeletionOrdering ordering(WriteExecutor executor) {
-            return TiebreakDeletionOrdering.EDGE;
+            return TiebreakDeletionOrdering.HAS;
         }
```
Comment (haikalpribadi) [inline graql/executor/property/HasAttributeExecutor.java:155]: This method takes in a `WriteExecutor` but not use it. Does the other implementation of the interface where this method comes from uses it properly? Some feels off...

File: graql/executor/property/RelationExecutor.java [hunk 2]
```diff
@@ -170,23 +164,27 @@ public VarProperty property() {
         }
 
         @Override
-        public Set<Variable> requiredVars() {
-            return allVars();
+        public TiebreakDeletionOrdering ordering(WriteExecutor executor) {
+            return TiebreakDeletionOrdering.ROLE_PLAYER;
+        }
```
Comment (haikalpribadi) [inline graql/executor/property/RelationExecutor.java:169]: Here as well...

Comment (flyingsilverfin) [reply graql/executor/property/RelationExecutor.java:169]: yeah only in 1 case do we actually need the WriteExecutor - with `isa`. We need to delete `Relation` instances before other types (and for now implicit attr relations are also deletable using `isa @has...`  for a short while

Comment (haikalpribadi) [reply graql/executor/property/RelationExecutor.java:169]: We can change it to not take in a `writeexecutor`?

**Pull request merged**